staging/src/k8s.io/client-go/tools/leaderelection/metrics.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leaderelection
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	leaderGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "leader_election_master_gauge",
```

Comment:
"leader_election_master_total" ? When unfiltered this metric does return the count of leaders in the system. If filtered by name then there is a 1 value for each component the reporter is master of, identified by name. ref https://prometheus.io/docs/practices/naming/

Reply:
+1 on `_total` instead of `_gauge`.

Reply:
After discussion further documenting the label. Higher level systems may tend to sum the results as mentioned however at this level it is 0/1 per label and am documenting as such. Also adding a warning that this should be grouped by label.